### ISSUE TITLE
Different order of emotion css files injected on local and production

### ISSUE THREAD

**Issue by mknmohit** [emotion-js/emotion#2764]
**Current behavior:**
I am currently using material-ui (MUI `v5.6.3`) along with emotion `v11` and tailwindcss `v3.0.15`.
Emotion style/CSS files should be injected above my custom styling files so that my custom styling can have more specificity.
I observed that the order of emotion style files injected into the head tag is different on local and different on production. On local, the emotion styling files are injected on top under the head tag, which seems to be correct. But on production  it is not working as expected.

I have taken screenshots of both environments.

#### Styling files order on Local
<img width="822" alt="Screenshot 2022-05-25 at 7 26 07 PM" src="https://user-images.githubusercontent.com/19211475/170440947-0f252541-d178-4cab-9234-ae330e65f82f.png">

#### Styling files order on Production
<img width="824" alt="Screenshot 2022-05-25 at 7 29 08 PM" src="https://user-images.githubusercontent.com/19211475/170440899-c325e7c1-1908-4aa6-824d-6abe987179fd.png">

I noticed one more thing If I locally build the production using `next build` and serve the prod build locally using `yarn start`, then the order of styling file is working fine, i.e. emotion styling files are getting injected at the top. But when I deploy the production build to the actual test/live environment, the order of styling files is not as expected.

**Expected behavior:**
The order of emotion styling/css files should be injected on top of my head tag or injected above my custom styling/css files so that the specificity of my custom styling will have more priority.

**Environment information:**

- `react` version: 17.0.2
- `@emotion/react` version: 11.9.0
- `@emotion/styled` version: 11.8.1
- `@emotion/server` version: 11.4.0
- `@mui/material` version: 5.6.3

**Comment by Andarist** [comment]
You can control where the styles should be injected by configuring `insertionPoint` on the cache

**Comment by srmagura** [comment]
Related issues:
- https://github.com/emotion-js/emotion/issues/2725
- https://github.com/emotion-js/emotion/issues/2741

@mknmohit What version of Next are you using? [This comment](https://github.com/emotion-js/emotion/issues/2725#issuecomment-1115492696) says their issue was fixed in Next 12.1.6. Though, I am not 100% sure if their issue is the same as your issue.

**Comment by StrikEr2909** [comment]
Next version is 12.1.0

**Comment by srmagura** [comment]
Try upgrading to to 12.1.6 and let us know if that fixes it.

**Comment by mknmohit** [comment]
Thanks for the response. I tried again after upgrading next to `v12.1.6` and it doesn't solve the problem 😞

**Comment by mknmohit** [comment]
> Can you try upgrading next to canary?

@siriwatknp I tried the `canary` version, and it also didn't work. 😢

**Comment by kachkaev** [comment]
A couple of own findings on this topic: https://github.com/mui/material-ui/issues/32916#issuecomment-1789913577

**TLDR:** I managed fix this problem by adding `_document.tsx` with `<meta name="emotion-insertion-point" content="" />`.

**Comment by felixelgato92** [comment]
I'm running into the same issue, and honestly this makes next.js awful for me. I'd rather just use a regular vanilla React app than deal with this non sense. I spent a few hours on a small project going crazy because my production build wasn't the same as the developer run. It also seems that similar issues have been going on for years, and they just seem to close out old ones, and put them into new issues so it has less votes.

**Comment by MetaMmodern** [comment]
Having same issue but without Next.js.
Our project uses material-ui(v4) and inside of it it used mui(v5) for one component with a custom mui theme.
Before it used webpack everything was fine. Now that I've update to use Vite the result is different in local build and deployment build. Which is wild, I should be able to repro the behaviour by doing a local build, but can't. The result of wrong css order is that that mui(v5) component doesn't have correct styles anymore.
